Makefile

```
@@ -1,4 +1,151 @@
-BASE_DIR ?= $(CURDIR)
+# Copyright 2021 TriggerMesh Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+KREPO      = triggermesh
+KREPO_DESC = TriggerMesh Open Source Components (sources and targets)
+
+
+BASE_DIR          ?= $(CURDIR)
+OUTPUT_DIR        ?= $(BASE_DIR)/_output
+
+# Dynamically generate the list of commands based on the directory name cited in the cmd directory
+COMMANDS          := $(shell ls $(BASE_DIR)/cmd)
```

A more Makefile'ish syntax for this is.

```suggestion
COMMANDS          := $(foreach dir,$(wildcard cmd/*),$(notdir $(dir)))
```

---

I hate Make 😅

---

Odd, an earlier attempt to use `notdir` ended up omitting the commands in the `cmd/` dir as they were in fact directories.

---

`notdir` just means "remove the directory from the path" (`cmd/` in this case).
It's a really terrible syntax, but AFAIK Make doesn't have a better built-in to do this (?)

---

Found out the `foreach` function is not required here so removed it from the suggestion.

---

Nice! Learned something.